ui/src/importer/importerTaskDetails/ImporterTaskDetails.js

```diff
@@ -40,28 +43,25 @@ export class ImporterTaskDetails extends React.Component {
   }
 
   checkForData = async () => {
-    const { importCompleted, data } = this.state;
-
-    if (importCompleted) {
+    if (this.importCompleted) {
       this.intervalId && clearInterval(this.intervalId);
       return;
     }
+
     try {
-      const nextEntry = _get(data, 'loaded_entries', 0);
-      const response = await importerApi.check(this.taskId, nextEntry);
-      if (response.data.status !== 'RUNNING') {
-        this.setState({
-          importCompleted: true,
-          isLoading: false,
-          data: response.data,
-        });
-      } else {
-        this.setState({
-          isLoading: true,
-          data: response.data,
-        });
-      }
+      this.requestBeingSent = true;
+
+      const { data } = await importerApi.check(this.taskId, 0);
```

Comment:
Checking more in details the code here, we need to use the `withCancel` pattern (see other components), to avoid memory leaks when using `this`.

Reply:
fair